orienteer-core/src/main/java/org/orienteer/core/method/IMethod.java

```diff
@@ -0,0 +1,29 @@
+package org.orienteer.core.method;
+
+import org.apache.wicket.Component;
+
+/**
+ * 
+ * Base interface for all methods
+ * If you need to use source method definition, annotate you class by {@link Method} annotation}
+ *
+ */
+public interface IMethod {
+	/**
+	 * Init method instance by environment data
+	 * Called only once per creation 
+	 * 
+	 * @param envData
+	 */
+	public void initialize(IMethodEnvironmentData envData);
```

Comment:
To avoid confusion, I recomment to rename it, because initialize in wicket is being used for a little bit other stuff.